website_sale_stock_list_preview/static/src/js/tour.js

```diff
@@ -53,6 +52,9 @@ odoo.define("website_sale_stock_list_preview.tour", function(require) {
             trigger: "a:contains('Test Product 7')",
             extra_trigger: ".text-danger:contains(' Temporarily out of stock')",
         },
+        {
+            trigger: "a[href='/shop']",
```

Comment:
This is probably not doing what you expect. It will get triggered before the previous step succeeds.

Add an `extra_trigger` that checks one element exists before hitting this link.

Reply:
This is just for return to the grid page. Have I to check an extra_trigger at all?

Reply:
Yes, actually looking at the logs, the tour is not working fine: https://travis-ci.org/github/OCA/e-commerce/jobs/691706160#L1606-L1620

It should display page reloads between steps, but there are none.

Imagine, let's describe how the previous step and this one should behave:

1. Wait until there's a `.text-danger` that contains "Temporarily out of stock".
1. Click on an `<a>` that contains "Test Product 7".
1. **Wait until the page for Test Product 7 is loaded.** (this is missing).
1. Click on the shop menu item.

As it isn't instructed to wait, it clicks the shop menu before the product page is loaded, so the tour isn't actually imitating what the user would do.